packages/better-auth/src/social-providers/patreon.ts

```diff
@@ -0,0 +1,103 @@
+import { betterFetch } from "@better-fetch/fetch";
+import {
+	createAuthorizationURL,
+	type OAuthProvider,
+	type ProviderOptions,
+	validateAuthorizationCode,
+} from "../oauth2";
+
+export interface PatreonProfile {
+	/**
+	 * The user id of the user
+	 */
+	id: string;
+
+	attributes: {
+		/**
+		 * The name of the user
+		 */
+		full_name: string;
+		/**
+		 * The email of the user
+		 */
+		email: string;
+		/**
+		 * The picture of the user
+		 */
+		image_url: string;
+		/**
+		 * If the email of the user is verified
+		 */
+		is_email_verified: boolean;
+	};
+}
+
+export interface PatreonOptions extends ProviderOptions<PatreonProfile> {}
+
+export const patreon = (options: PatreonOptions) => {
+	return {
+		id: "patreon",
+		name: "Patreon",
+		createAuthorizationURL({ state, scopes, redirectURI, codeVerifier }) {
+			const _scopes = options.disableDefaultScope ? [] : ["identity[email]"];
```

Comment:
can we have support for a custom `scope` array via PatreonOptions? I basically would need "identity identity[email]".

Thanks for this PR btw